### PR TITLE
Mention Kotlin DSL faster first use in release notes

### PR DESCRIPTION
**Pull request by eskatos**
* Fixes https://github.com/gradle/gradle/issues/26483
* Follows up https://github.com/gradle/gradle/pull/26620

----

Rendered release notes
https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/72743635:id/distributions/gradle-8.5-docs.zip!/gradle-8.5-20231004142122%2B0000/docs/release-notes.html#kotlin-dsl-improvements

---

The inline image is:

kotlin-dsl-faster-first-use

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test BD

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -88,6 +88,20 @@ The JAR file is now self-attributing and you don't need to consider whether you
 
 Gradle's [Kotlin DSL](userguide/kotlin_dsl.html) provides an enhanced editing experience in supported IDEs compared to the traditional Groovy DSL — auto-completion, smart content assist, quick access to documentation, navigation to source, and context-aware refactoring.
 
+#### Faster first use
+
+With an empty [Gradle User Home](userguide/directory_layout.html#dir:gradle_user_home), before compiling the first `.gradle.kts` script, Gradle generates the Gradle API Kotlin DSL Extensions JAR.
```
Comment (jbartok) [inline subprojects/docs/src/docs/release/notes.md:93]: Feels a bit "abrupt", like it needs a gentler introduction, more context, not exactly sure what.

Something like "when running a build with an empty Gradle user home (for example first install, ephemeral CI, whatnot)", before we compile ....

Take the next sentence and put it first maybe?

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -88,6 +88,20 @@ The JAR file is now self-attributing and you don't need to consider whether you
...
+This happens on every developer machine or CI agent, most often with a cold [Gradle daemon](userguide/gradle_daemon.html).
+This is quite a resource intensive process which accounts for around 4 seconds on beefy machines ; much more on slower machines.
+
+Starting with this release, the Kotlin extensions for the Gradle API are now shipped at part of the Gradle distribution.
```
Comment (jbartok) [inline subprojects/docs/src/docs/release/notes.md:97]:
```suggestion
Starting with this release, the Kotlin extensions for the Gradle API are now shipped as part of the Gradle distribution.
```

**Review by jbartok — COMMENTED** [review]
Just a couple of comments.

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -88,6 +88,20 @@ The JAR file is now self-attributing and you don't need to consider whether you
 
 Gradle's [Kotlin DSL](userguide/kotlin_dsl.html) provides an enhanced editing experience in supported IDEs compared to the traditional Groovy DSL — auto-completion, smart content assist, quick access to documentation, navigation to source, and context-aware refactoring.
 
+#### Faster first use
+
+With an empty [Gradle User Home](userguide/directory_layout.html#dir:gradle_user_home), before compiling the first `.gradle.kts` script, Gradle generates the Gradle API Kotlin DSL Extensions JAR.
+This happens on every developer machine or CI agent, most often with a cold [Gradle daemon](userguide/gradle_daemon.html).
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:94]:
```suggestion
This happens on every developer machine or CI agent, often with a cold [Gradle daemon](userguide/gradle_daemon.html).
```

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -88,6 +88,20 @@ The JAR file is now self-attributing and you don't need to consider whether you
...
+This is quite a resource intensive process which accounts for around 4 seconds on beefy machines ; much more on slower machines.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:95]:
```suggestion
This resource-intensive process takes around 4 seconds on powerful machines and is much longer on slower machines.
```

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:93]:
```suggestion
Gradle generates a Gradle API Kotlin DSL Extensions JAR before compiling the first `.gradle.kts` script when the [Gradle User Home](userguide/directory_layout.html#dir:gradle_user_home) is empty.
```

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:97]:
```suggestion
The Kotlin extensions for the Gradle API now ship as part of the Gradle distribution.
```

File: subprojects/docs/src/docs/release/notes.md [hunk 5]
```diff
@@ -88,6 +88,20 @@ The JAR file is now self-attributing and you don't need to consider whether you
...
+
+Starting with this release, the Kotlin extensions for the Gradle API are now shipped at part of the Gradle distribution.
+The consequence is that first-use of a Gradle version for compiling `.gradle.kts` scripts is faster.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:98]:
```suggestion
Consequently, the first use of a Gradle version for compiling `.gradle.kts` scripts is much faster.
```

**Review by lkasso — APPROVED** [review]
Minor changes to release notes.

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (eskatos) [reply subprojects/docs/src/docs/release/notes.md]: :+1: incorporated manually given conflicts

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (eskatos) [reply subprojects/docs/src/docs/release/notes.md]: Agreed. Does it feels better now with Laura's suggestions applied?

**Comment by eskatos** [comment]
Thanks a lot @lkasso!

File: subprojects/docs/src/docs/release/notes.md [hunk 6]
```diff
@@ -88,6 +88,20 @@ The JAR file is now self-attributing and you don't need to consider whether you
 
 Gradle's [Kotlin DSL](userguide/kotlin_dsl.html) provides an enhanced editing experience in supported IDEs compared to the traditional Groovy DSL — auto-completion, smart content assist, quick access to documentation, navigation to source, and context-aware refactoring.
 
+#### Faster first use
+
+Gradle generates a Gradle API Kotlin DSL Extensions JAR before compiling the first `.gradle.kts` script when the [Gradle User Home](userguide/directory_layout.html#dir:gradle_user_home) is empty.
+This happens on every developer machine or CI agent, often with a cold [Gradle daemon](userguide/gradle_daemon.html).
```
Comment (jbartok) [inline subprojects/docs/src/docs/release/notes.md:94]: The "often with a cold daemon" statement seems misleading. This is not about the daemon being cold, but about the gradle user home not having stuff.

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (jbartok) [reply subprojects/docs/src/docs/release/notes.md]: Definitely.

File: subprojects/docs/src/docs/release/notes.md (hunk 6 above)
Comment (eskatos) [reply subprojects/docs/src/docs/release/notes.md:94]: It adds up. 99% of the time you'll have both a cold daemon and empty user home, making this even slower.

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**